spec/rubocop/cop/rspec/string_as_instance_double_constant_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::StringAsInstanceDoubleConstant,
+               :config do
+  context 'when using a class for instance_double' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        instance_double(Foo, failure_message, stubs)
+      RUBY
+    end
+  end
+
+  context 'when using a symbol for instance_double' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        instance_double(:Foo, failure_message, stubs)
+      RUBY
+    end
+  end
+
+  context 'when using a string for instance_double' do
+    it 'replaces the string with the class' do
+      expect_offense <<~RUBY
+        instance_double('Foo', failure_message, stubs)
+                        ^^^^^ Do not use a string as `instance_double` constant.
+      RUBY
+
+      expect_correction <<~RUBY
+        instance_double(Foo, failure_message, stubs)
+      RUBY
+    end
+  end
+end
```

Comment:
Please add a case when the first argument is a variable/method `instance_double(type_undetectable_in_static_analysis)`.

Reply:
Thanks, done!